### PR TITLE
Remove qiskit.org links

### PR DESCRIPTION
**Pull request by Eric-Arellano**

### CODE REVIEW

File: docs/tutorials/Rabi_oscillations.rst [hunk 1]
```diff
@@ -21,9 +21,11 @@ Hamiltonian containing a diagonal term of frequency :math:`\nu_z` and a
 transverse term of amplitude :math:`\nu_x` and harmonic driving
 frequency :math:`\nu_d` (see how the Hamiltonians are derived on 
 `Qiskit Textbook page on Introduction to Transmon Physics
-<https://qiskit.org/textbook/ch-quantum-hardware/transmon-physics.html>`_),
+<https://github.com/Qiskit/platypus/blob/main/notebooks/v2/quantum-hardware-pulses/transmon-physics.ipynb>`_),
 
-.. math:: H = \frac{1}{2} \times 2 \pi \nu_z {Z} + 2 \pi \nu_x \cos(2 \pi \nu_d t){X},
```
Comment (Eric-Arellano) [inline docs/tutorials/Rabi_oscillations.rst:26]: This didn't render the math properly.

File: docs/tutorials/Rabi_oscillations.rst [hunk 2]
```diff
@@ -21,9 +21,11 @@ Hamiltonian containing a diagonal term of frequency :math:`\nu_z` and a
...
+.. math::
+
+    H = \frac{1}{2} \times 2 \pi \nu_z {Z} + 2 \pi \nu_x \cos(2 \pi \nu_d t){X},
```
Comment (Eric-Arellano) [inline docs/tutorials/Rabi_oscillations.rst:28]: Is the comma at the end a bug?

File: docs/userguide/how_to_use_pulse_schedule_for_jax_jit.rst [hunk 3]
```diff
@@ -6,7 +6,7 @@ How-to use pulse schedules generated by ``qiskit-pulse`` with JAX transformation
 ``Qiskit-pulse`` enables specification of time-dependence in quantum systems as pulse schedules,
 built from sequences of a variety of instructions, including the specification of shaped pulses (see
 the detailed  API information about `Qiskit pulse API Reference
-<https://qiskit.org/documentation/apidoc/pulse.html>`__). As of ``qiskit-terra`` 0.23.0, JAX support
+<https://docs.quantum.ibm.com/api/qiskit/pulse>`__). As of ``qiskit`` 0.40.0, JAX support
```
Comment (Eric-Arellano) [inline docs/userguide/how_to_use_pulse_schedule_for_jax_jit.rst:9]: We don't talk about Qiskit Terra anymore.

File: releasenotes/notes/0.4/move-repo-c0b48ba3b0ced8db.yaml [hunk 4]
```diff
@@ -4,6 +4,6 @@ other:
     The repository has been moved from 
     [github.com/Qiskit/qiskit-dynamics](https://github.com/Qiskit/qiskit-dynamics) to 
     [github.com/Qiskit-Extensions/qiskit-dynamics](https://github.com/Qiskit-Extensions/qiskit-dynamics), 
-    and the documentation has been moved from
-    [qiskit.org/documentation/dynamics](https://qiskit.org/documentation/dynamics) to 
-    [qiskit.org/ecosystem/dynamics](https://qiskit.org/ecosystem/dynamics/).
+    and the [documentation](https://qiskit-extensions.github.io/qiskit-dynamics/) has been moved from
+    qiskit.org/documentation/dynamics
+    qiskit.org/ecosystem/dynamics
```
Comment (Eric-Arellano) [inline releasenotes/notes/0.4/move-repo-c0b48ba3b0ced8db.yaml:9]: No point linking to the old sites since they redirect to the new location anyways.

**Review by DanPuzzuoli — APPROVED** [review]

**Pull request merged**